internal/provider/resource_source_test.go

```diff
@@ -194,6 +195,14 @@ func TestResourceSource(t *testing.T) {
 					ingesting_paused  = true
 					scrape_urls      = ["http://localhost:9100/metrics"]
 					scrape_frequency_secs = 30
+					scrape_request_basic_auth_user = "user1"
+					scrape_request_basic_auth_password = "password1"
+					scrape_request_headers = [
+						{ 
+							name = "Authorization",
+							value = "Bearer foo"
+						}
+					]
```

Comment:
💄 Since you've added comprehensive test of scraping below, I would revert these changes to keep the test simple - WDYT? 

Adding the rest of removals as suggestions for easy update if you decide to do so 👍 

```suggestion
```

Reply:
I didn't intend to capture all the other scrape request attributes checks in the header tests. I'll simplify the header check tests to just check that, and leave the existing test as a "set all the attributes" test.